test/module/irohad/simulator/simulator_test.cpp

```diff
@@ -168,30 +168,18 @@ TEST_F(SimulatorTest, ValidWhenPreviousBlock) {
 
   init();
 
-  auto proposal_wrapper =
-      make_test_subscriber<CallExact>(simulator->onVerifiedProposal(), 1);
```

Comment:
There are no tests which check the successful branch in the subscription lambda created in constructor, so the coverage is reduced.
I would suggest to either: 
- add a separate test case with `  auto proposal_wrapper = make_test_subscriber<CallExact>(simulator->onVerifiedProposal(), 1);`, so that successful branch is covered.
- or rewrite the test with `subject<OrderingEvent>`, create `Simulator` in setup and subscribe it to the subject, remove init method, leave the wrappers in modified tests, and trigger the processing by calling `subject.get_subscriber().on_next()`. This approach is used in `synchronizer_test`